python/ccxt/async_support/base/exchange.py

```diff
@@ -134,7 +137,7 @@ async def fetch(self, url, method='GET', headers=None, body=None):
             url = proxyUrl + url
         # proxy agents
         final_proxy = None  # set default
-        final_session = None
+        self.socks_proxy_session = None
```

Comment:
I don't think this is correct - but should be wrapped in a conditional.

Unconditionally setting this to `None` will lose reference to the previous `ClientSession` - which is what's most likely causing the "unclosed client session" messages i'm seeing.